src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -482,6 +482,12 @@ MultiLogicalPlanOptimize(MultiTreeRoot *multiLogicalPlan)
 						errhint("You might need to disable approximations for either "
 								"count(distinct) or limit through configuration.")));
 	}
+
+	if (TargetListContainsSubquery(masterExtendedOpNode->targetList))
```

Comment:
While testing this patch, the error message was the challenge as there are no clues why the query is not pushed down.


A this point it seems hard to spot the exact cause for the error. But, can we at least hint the user about what are the supported structure?